libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -3167,6 +3167,23 @@ bool IRGeneratorForStatements::visit(TryCatchClause const& _clause)
 void IRGeneratorForStatements::setLocation(ASTNode const& _node)
 {
 	m_currentLocation = _node.location();
+	appendCurrentLocation();
+}
+
+void IRGeneratorForStatements::appendCurrentLocation()
+{
+	if (m_currentLocation.isValid() && m_code.tellp() != m_lastSourceLocationPosition)
```

Comment:
not sure if using `tellp` is the best idea how to avoid multiple consecutive writes of source location, but it was the first idea I got.

Reply:
Is `setLocation` actually called so many repeated times where it should not be? I think we can just go with putting this out in `setLocation` unconditionally and call it a day.